packages/myst-directives/src/code.ts

```diff
@@ -1,56 +1,83 @@
 import type { Caption, Container } from 'myst-spec';
 import type { Code } from 'myst-spec-ext';
 import yaml from 'js-yaml';
-import type { DirectiveSpec, GenericNode } from 'myst-common';
+import type { DirectiveData, DirectiveSpec, GenericNode } from 'myst-common';
 import { fileError, fileWarn, normalizeLabel, ParseTypesEnum } from 'myst-common';
+import type { VFile } from 'vfile';
 
-export const codeDirective: DirectiveSpec = {
-  name: 'code',
-  arg: {
-    type: ParseTypesEnum.string,
+function parseEmphasizeLines(emphasizeLinesString?: string | undefined): number[] | undefined {
+  if (!emphasizeLinesString) return undefined;
+  const emphasizeLines = emphasizeLinesString
+    ?.split(',')
+    .map((val) => Number(val.trim()))
+    .filter((val) => Number.isInteger(val));
+  return emphasizeLines;
+}
+
+/** This function parses both sphinx and RST code-block options */
+export function getCodeBlockOptions(
+  options: DirectiveData['options'],
+  vfile: VFile,
+): Pick<Code, 'emphasizeLines' | 'showLineNumbers' | 'startingLineNumber'> {
+  if (options?.['lineno-start'] != null && options?.['number-lines'] != null) {
+    fileWarn(vfile, 'Cannot use both "lineno-start" and "number-lines"', {
+      source: 'code-cell:tags',
+    });
+  }
+  const emphasizeLines = parseEmphasizeLines(options?.['emphasize-lines'] as string | undefined);
+  const numberLines = options?.['number-lines'] as number | undefined;
+  // Only include this in mdast if it is `true`
+  const showLineNumbers =
+    options?.linenos || options?.['lineno-start'] || options?.['lineno-match'] || numberLines
```

Comment:
This fixes a portion of the bug in:
- https://github.com/executablebooks/jupyterlab-myst/issues/174

Line number is now turned on if `lineno-start` is included.